File: php/word_count.php
<?php

function wordCount(string $phrase) : array {
   $phrase = prepareString($phrase);
   $words = explode(' ', $phrase);  

   array_walk($words, function ($word) use (&$result) {
      if (empty($result[$word])) {
         $result[$word] = 1;
      }
      else {
         $result[$word]++;
      }
   });

   return $result;
}

function prepareString(string $phrase) : string {
   $phrase = trim($phrase);
   $phrase = strtolower($phrase);
   $phrase = preg_replace('/[^a-z0-9\s]/', '', $phrase);
   $phrase = preg_replace('/(\v|\s)+/', ' ', $phrase);
   return $phrase;
}

File: python/word_count.py
import re;

def word_count(phrase):
    result = {}
    phrase = re.sub('[\s+_,]', ' ', phrase.lower())
    words = re.sub('[^a-z0-9\s]', '', phrase).split()
    for word in range(0,len(words)):
       count = len(re.findall("\\b" + re.escape(words[word]) + "\\b", phrase))
       result[words[word]] = count;
    return result

